apps/mobile/src/features/ConfirmTx/components/SignTransaction/SignTransaction.tsx

```diff
@@ -6,27 +6,44 @@ import { selectChainById } from '@/src/store/chains'
 import { RootState } from '@/src/store'
 import type { ChainInfo } from '@safe-global/safe-gateway-typescript-sdk'
 import { useLocalSearchParams } from 'expo-router'
-import { useLazySignTransactionQuery } from '@/src/store/signersBalance'
 import { getPrivateKey } from '@/src/hooks/useSign/useSign'
 import SignError from './SignError'
 import SignSuccess from './SignSuccess'
+import { signTx } from '@/src/services/tx/tx-sender/sign'
+import { useTransactionsAddConfirmationV1Mutation } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import logger from '@/src/utils/logger'
 
 export function SignTransaction() {
   const { txId, signerAddress } = useLocalSearchParams<{ txId: string; signerAddress: string }>()
   const activeSafe = useDefinedActiveSafe()
   const activeChain = useAppSelector((state: RootState) => selectChainById(state, activeSafe.chainId))
 
-  const [trigger, { isFetching, data, isError }] = useLazySignTransactionQuery()
+  const [addConfirmation, { isLoading, data, isError }] = useTransactionsAddConfirmationV1Mutation()
 
   const sign = useCallback(async () => {
-    const privateKey = await getPrivateKey(signerAddress)
-
-    trigger({
-      chain: activeChain as ChainInfo,
-      activeSafe,
-      txId,
-      privateKey,
-    })
+    try {
+      const privateKey = await getPrivateKey(signerAddress)
+
+      const signedTx = await signTx({
+        chain: activeChain as ChainInfo,
+        activeSafe,
+        txId,
+        privateKey,
+      })
+
+      // Now call the mutation with the signed transaction data
+      await addConfirmation({
+        chainId: activeSafe.chainId,
+        safeTxHash: signedTx.safeTransactionHash,
+        addConfirmationDto: {
+          // TODO: we need to add this signature type in the auto generated types, because it was included recently in CGW
+          // @ts-ignore
+          signature: signedTx.signature,
+        },
+      })
+    } catch (error) {
+      logger.error('Error signing transaction:', error)
+    }
   }, [activeChain, activeSafe, txId, signerAddress])
 
   useLayoutEffect(() => {
```

Comment:
Not a mobile dev but why is this happening in an effect? Is there an [event handler that can be used](0x6e4c29936929629df7af5986078b71f879c1694a9bba03c2f983d4ff21bc483a695f04341b1f1533d4420fb5b91f13ea66291901197d20ac19420a5f81d70d151f)?

Reply:
@iamacook this is done in a effect because this transaction is done in a "loading screen" (you can see it in the video). So there is no explicit trigger for this call in the screen.

Reply:
Is it possible to do it when pressing "Confirm"? It's generally [good practice](https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers) to avoid event-specific logic in an effect.

Reply:
yeah I know, the thing is that when pressing confirm we change the screen of the app, its like going from one page to another in a web project, so I lose context of everything including the `data` which is returned from the mutation.